h2/src/main/org/h2/table/Table.java

```diff
@@ -620,7 +620,7 @@ public void dropMultipleColumnsConstraintsAndIndexes(Session session,
     }
 
     public Row getTemplateRow() {
-        return database.createRow(new Value[columns.length], Row.MEMORY_CALCULATE);
+        return database.createRow(new Value[getColumns().length], Row.MEMORY_CALCULATE);
```

Comment:
this is a little icky, it means that the columns field no longer always applies to the Table class. Probably better to create an AbstractTable class to act as the common base class for this and  TableSynonym